controllers/selfnoderemediationconfig_controller.go

```diff
@@ -190,3 +198,54 @@ func (r *SelfNodeRemediationConfigReconciler) syncCerts(cr *selfnoderemediationv
 	}
 	return nil
 }
+
+func (r *SelfNodeRemediationConfigReconciler) updateDsTolerations(objs []*unstructured.Unstructured, tolerations []corev1.Toleration) error {
+	r.Log.Info("Updating DS tolerations")
+	if len(tolerations) == 0 {
+		return nil
+	}
+	for _, toleration := range tolerations {
+		for _, ds := range objs {
+			if err := r.updateTolerationOnDs(ds, toleration); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (r *SelfNodeRemediationConfigReconciler) updateTolerationOnDs(ds *unstructured.Unstructured, toleration corev1.Toleration) error {
+	tolerations, _, err := unstructured.NestedFieldNoCopy(ds.Object, "spec", "template", "spec", "tolerations")
+	if err != nil {
+		r.Log.Error(err, "tolerations not found in ds")
+		return err
+	}
+
+	updatedTolerations, ok := tolerations.([]interface{})
+	if !ok {
+		err := fmt.Errorf(fmt.Sprintf("failed to convert tolerations expected of type %T but got %T", []interface{}{}, tolerations))
+		r.Log.Error(err, err.Error())
+		return err
+	}
+	r.Log.Info("updateTolerationOnDs original tolerations", "tolerations", updatedTolerations)
+
+	configTolerations := map[string]interface{}{
```

Comment:
- `configTolerations` isn't a very descriptive name, is it? 🤔 
- this conversion can be done easier: `newToleration, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&toleration)`